core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4891,20 +4887,12 @@ else if(displayName.equals(item.getDisplayName())) {
      */
     boolean isNameUnique(String name, String currentJobName) {
         Item item = getItem(name);
-
+        
         if(null==item) {
             // the candidate name didn't return any items so the name is unique
             return true;
         }
-        else if(item.getName().equals(currentJobName)) {
-            // the candidate name returned an item, but the item is the item
-            // that the user is configuring so this is ok
-            return true;
-        }
-        else {
-            // the candidate name returned an item, so it is not unique
-            return false;
-        }
+        else return item.getName().equals(currentJobName);
```

Comment:
The whole `if` here could be simplified to `return item == null || item.getName().equals(currentJobName);`

Reply:
I prefer the more verbose code because of the comments.  Maintaining the code is more difficult when clear comments like this are deleted.

Reply:
The `else` can/should also be removed. It doesn't serve any purpose since this is already similar to a return-early pattern.

Reply:
Either keep the curly brackets
```suggestion
        else {
            return item.getName().equals(currentJobName);
        }    
```

Or

```suggestion
        return item.getName().equals(currentJobName);
```

Would be nice :)